### ISSUE TITLE
When launched with tsx, process.on('message') and process.send doesn't work

### ISSUE THREAD

**Issue by FishingHacks** [privatenumber/tsx#201]
### Bug description

I launched tsx with my ts-file, using `child_process` and with ipc attached (`spawn('tsx', ["<filepath>"], {stdio: ["ipc"]})`).
I then listened for messages, and sent a message to the process, which the process should respond to, and then exit, but instead, nothing happened. It doesn't throw an error either. I hope this description is clear enough.

### Reproduction

https://github.com/FishingHacks/tsx-error-ipc

Requirements:
- Globally installed tsx
- Globally installed ts-node (it will launch `b.ts` with ts-node as a comparison, that code-part is not neccessary)

### Environment

```shell
System:
    OS: Linux 5.4 Linux Mint 20.3 (Una)
    CPU: (4) x64 Intel(R) Core(TM) i5-3450 CPU @ 3.10GHz
    Memory: 11.36 GB / 15.57 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 18.6.0 - /usr/bin/node
    npm: 9.4.0 - /usr/bin/npm
```

### Can you work on a fix?

- [ ] I’m interested in opening a pull request to address this issue.

**Comment by FishingHacks** [comment]
it appears that 
```typescript
if (process.send) {
    let handledDep = false;
    childProcess.on('message', (message: any) => {
        if (message && message.type === 'dependency') handledDep = true;
        else if (handledDep && (!message || message.type !== 'kill'))
            process.send?.(message);
    });
    process.on('message', (msg: any) => childProcess.send(msg));
}
```
after the relaySignal function definition, it essentially fixes the error. Then i noticed, that `process.off` does not work, but `process.removeListener` does, so i added `process.off = process.removeListener;` at the end of `preflight.cts`.

When i tried to run `npm test`, it "froze" at `✔ Package type: commonjs › tsx › watch › ignore › file path & glob`, but that does not appear to be caused by the changes i made, as reverting them also displayed the same behavior.

**Comment by privatenumber** [comment]
tsx is a wrapper process, and I'm not sure if I recall relaying IPC from the actual Node.js process to the parent. I think this will have to be researched & implemented.

I'm not sure if I understand the point of your last message—what are you trying to do?

**Comment by alanhamlett** [comment]
Looks like an attempt was made to fix this in #206, but it's been a while and that PR seems stalled.
